src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -299,6 +299,17 @@ class ConstraintsChecker
         return CheckConstraintMinValue(itemName, current.Value(), static_cast<T>(expected));
     }
 
+    template <typename T, typename U, std::enable_if_t<!std::is_pointer<U>::value, int> = 0>
```

Comment:
One last question: why is the `!std::is_pointer<U>` bit needed here?

Reply:
Sorry, I think that was a hangover from before when I didn't have U as a reference. Might have been wrong there too, sorry, quite new to templating, but I don't think we need it.